space_packet_parser/xtce/validation.py

```diff
@@ -0,0 +1,441 @@
+        if self.column_number is not None:
+            location_parts.append(f"col {self.column_number}")
+        if self.xpath_location:
+            location_parts.append(f"xpath: {self.xpath_location}")
+
+        location_str = f" ({', '.join(location_parts)})" if location_parts else ""
+        return f"ERROR: {self.message}{location_str}"
+
+
+@dataclass
+class ValidationResult:
+    """Results of XTCE document validation."""
+
+    valid: bool
+    validation_level: ValidationLevel
+    errors: list[ValidationError] = field(default_factory=list)
+    schema_version: Optional[str] = None
+    schema_location: Optional[str] = None
+    validation_time_ms: Optional[float] = None
+
+    def add_error(
+        self,
+        message: str,
+        error_code: str,
+        xpath_location: Optional[str] = None,
+        line_number: Optional[int] = None,
+        context: Optional[dict[str, Any]] = None,
+    ):
+        """Add a validation error."""
+        error = ValidationError(
+            message=message,
+            error_code=error_code,
+            xpath_location=xpath_location,
+            line_number=line_number,
+            context=context or {},
+        )
+        self.errors.append(error)
+        self.valid = False
+
+    def __str__(self) -> str:
+        """String representation of validation result."""
+        status = "VALID" if self.valid else "INVALID"
+        result = f"Validation Result: {status} ({self.validation_level.value} level)\n"
+
+        if self.errors:
+            result += f"\nErrors ({len(self.errors)}):\n"
+            for error in self.errors:
+                result += f"  {error}\n"
+
+        return result
+
+
+class XtceValidationError(Exception):
+    """Exception raised during XTCE validation."""
+
+    def __init__(self, message: str, validation_result: Optional[ValidationResult] = None):
+        super().__init__(message)
+        self.validation_result = validation_result
+
+
+def _fix_known_schema_issues(schema_content: str) -> str:
+    """Fix known issues in the official XTCE XSD schema.
+
+    The official OMG XTCE schema references xml:base but doesn't declare
+    the xml namespace, causing lxml validation to fail.
+    """
+    if 'ref="xml:base"' in schema_content:
+        import re
+
+        # Remove the problematic reference entirely since it's optional for validation
+        schema_content = re.sub(
+            r'\s*<attribute\s+ref="xml:base"\s*/>\s*',
+            "\n\t\t\t\t<!-- xml:base attribute removed for lxml compatibility -->\n\t\t\t\t",
+            schema_content,
+        )
+        schema_content = re.sub(
+            r'\s*<attribute\s+ref="xml:base"></attribute>\s*',
+            "\n\t\t\t\t<!-- xml:base attribute removed for lxml compatibility -->\n\t\t\t\t",
+            schema_content,
+        )
+    return schema_content
+
+
+def _load_schema(schema_url: str, timeout: int = 30) -> tuple[ElementTree.XMLSchema, str]:
```

Comment:
Is this something worth shipping with the package itself? Download it and redistribute it so you can reference it locally rather than having to go retrieve it.